PatchesSystem/AsmPatch.cs

```diff
@@ -0,0 +1,189 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ZeldaFullEditor.Data
+{
+    /// <summary>
+    ///     Class used to load ASM file as data
+    /// </summary>
+    internal class AsmPatch
+    {
+        /// <summary>
+        ///     Contains all the defines informations for the patch
+        /// </summary>
+        internal Dictionary<string, Dictionary<string, string>> PatchDefines = new Dictionary<string, Dictionary<string, string>>();
+
+        /// <summary>
+        ///     Gets or sets name of the patch that appears inside of the asm file
+        /// </summary>
+        public string PatchName { get; set; }
+
+        /// <summary>
+        ///     Version of the patch that appears inside of the asm file
+        /// </summary>
+        internal string PatchVersion;
+
+        /// <summary>
+        ///     Author of the patch that appears inside of the asm file
+        /// </summary>
+        internal string PatchAuthor;
+
+        /// <summary>
+        ///     Description of the patch that appears inside of the asm file
+        /// </summary>
+        internal string PatchDescription;
+
+        /// <summary>
+        ///     The entire patch as string
+        /// </summary>
+        internal string WholePatch;
+
+        /// <summary>
+        ///     The folder the patch has been loaded from
+        /// </summary>
+        internal string PatchFolder;
+
+        /// <summary>
+        ///     The filename only the patch has been loaded from
+        /// </summary>
+        internal string FileName;
+
+        /// <summary>
+        ///     Gets or sets a value indicating whether the folder the patch has been loaded from
+        /// </summary>
+        public bool PatchEnabled { get; set; } = true;
+
+        public AsmPatch(string filename, string subfolder)
```

Comment:
nit: `string filepath` instead of filename

Reply:
done.